File: source/Team1.Model/Enums/GalleryTypeEnum.cs
using System.ComponentModel.DataAnnotations;

namespace Team1.Model.Enums
{
    public enum GalleryTypeEnum : byte
    {
        Public = 1,
        [Display(Name = "My Gallery")]
        MyGallery = 2,
        [Display(Name = "Admin Mode")]
        AdminMode = 3,
    }
}

File: source/Team1.Web.Site/ClientApp/src/app.common/enums/GalleryTypeEnum.ts



import {ISelectOption} from 'app.common/dtos/SelectOptionDto';
enum GalleryTypeEnum {
        Public = 1,
        MyGallery = 2,
        AdminMode = 3,
         
}

export class GalleryTypeEnum_class {

public static enum = GalleryTypeEnum;

  public static readonly enumByText : { [key : string ]: number} = {
    ['Public']: 1,
    ['MyGallery']: 2,
    ['AdminMode']: 3,
    
  };
  public static readonly enumByNumber : { [key : number ]: string} = {
    [1]: "Public",
    [2]: "My Gallery",
    [3]: "Admin Mode",
    
  };

  public static readonly enumAsSelectOptions:Array<ISelectOption<number>> = [
  {value: 1, text: "Public"},
  {value: 2, text: "My Gallery"},
  {value: 3, text: "Admin Mode"},
  ]
}

